lib/compiler-llvm/src/translator/code.rs

```diff
@@ -3835,21 +3942,15 @@ impl<'ctx, 'a> LLVMFunctionCodeGenerator<'ctx, 'a> {
                 //       v1 < v2 or
                 //       v2 is ±zero and v1 is negative.
 
-                let ((v1, i1), (v2, i2)) = self.state.pop2_extra()?;
-                let (v1, i1) = self.v128_into_f32x4(v1, i1);
-                let (v2, i2) = self.v128_into_f32x4(v2, i2);
-                let v1 = if !i1.is_arithmetic_f32() {
-                    self.canonicalize_nans(v1.as_basic_value_enum())
-                        .into_vector_value()
-                } else {
-                    v1
-                };
-                let v2 = if !i2.is_arithmetic_f32() {
-                    self.canonicalize_nans(v2.as_basic_value_enum())
-                        .into_vector_value()
-                } else {
-                    v2
-                };
+                let (v1, v2) = self.state.pop2()?;
+                let v1 = self.v128_into_f32x4(v1);
+                let v2 = self.v128_into_f32x4(v2);
+                let v1 = self
+                    .canonicalize_nans(v1.as_basic_value_enum())
+                    .into_vector_value();
+                let v2 = self
+                    .canonicalize_nans(v2.as_basic_value_enum())
+                    .into_vector_value();
 
                 let v1_is_nan = self.builder.build_float_compare(
```

Comment:
This much be replaced with a call to `llvm.experimental.constrained.fcmp`:
> If any FP operation in a function is constrained then they all must be constrained. This is required for correct LLVM IR.
-- https://llvm.org/docs/LangRef.html#constrained-floating-point-intrinsics

Please audit for any other non-constrainted floating point operations.